fawltydeps/packages.py

```diff
@@ -131,10 +136,18 @@ def packages(self) -> Dict[str, Package]:
         This enumerates the available packages  _once_, and caches the result for
         the remainder of this object's life in _packages.
         """
+        user_mapping: TomlData = {}
+
+        if self.mapping_path is not None:
+            logger.debug(f"Loading user-defined mapping from {self.mapping_path}")
+            with open(self.mapping_path, "rb") as mapping_file:
+                toml_data = tomllib.load(mapping_file)
+                user_mapping = {**user_mapping, **toml_data}
 
-        logger.debug(f"Loading user-defined mapping from {self.mapping_path}")
-        with open(self.mapping_path, "rb") as mapping_file:
-            user_mapping = tomllib.load(mapping_file)
+        if self.custom_mapping is not None:
+            logger.debug("Loading user-defined mapping from custom mapping.")
+            # Custom mapping overrides mapping from file
+            user_mapping = {**user_mapping, **self.custom_mapping}
```

Comment:
As per your comment above, and the way the dict is constructed here, I assume that if a package mapping is defined twice, the mapping from the toml section will override the mapping from file. But this is not visible to the user. When there's a mismatch between the two, I would expect:
- either an error to be thrown, informing the user that there's a mismatch between what they defined in the config section and what they defined in the file and that they need to fix that.
- or a union to be performed between the two, while also log.info-rming the user that we are performing a union.
What do you think?

Reply:
So let's go with the union approach. I added also the info message for cases, where the same name is encountered in mapping from file and from configuration. I also added test for this interaction in `test_packages`.

Reply:
That looks great!